File: sixth-exercises/python/5.py
def invert_and_replace(input_string, char_to_replace='A', replacement_char='*'):
    reversed_string = input_string[::-1]
    replaced_string = reversed_string.replace(char_to_replace, replacement_char)
    return replaced_string

phrase1 = input("Digite a primeira frase: ")
phrase2 = input("Digite a segunda frase: ")

inverted_replaced_phrase1 = invert_and_replace(phrase1)
inverted_replaced_phrase2 = invert_and_replace(phrase2)

print("Frase 1 invertida e com 'A' substituídos por '*':", inverted_replaced_phrase1)
print("Frase 2 invertida e com 'A' substituídos por '*':", inverted_replaced_phrase2)

File: sixth-exercises/c/5.c
#include <stdio.h>
#include <string.h>

void reverse_and_replace(char *input_string, char char_to_replace, char replacement_char) {
    int len = strlen(input_string);
    for (int i = 0; i < len; i++) {
        if (input_string[i] == char_to_replace) {
            input_string[i] = replacement_char;
        }
    }

    for (int i = 0, j = len - 1; i < j; i++, j--) {
        char temp = input_string[i];
        input_string[i] = input_string[j];
        input_string[j] = temp;
    }
}

int main() {
    char phrase1[100];
    char phrase2[100];

    printf("Digite a primeira frase: ");
    fgets(phrase1, sizeof(phrase1), stdin);
    printf("Digite a segunda frase: ");
    fgets(phrase2, sizeof(phrase2), stdin);

    reverse_and_replace(phrase1, 'A', '*');
    reverse_and_replace(phrase2, 'A', '*');

    printf("Frase 1 invertida e com 'A' substituídos por '*': %s\n", phrase1);
    printf("Frase 2 invertida e com 'A' substituídos por '*': %s\n", phrase2);

    return 0;
}
